code/datums/elements/lazystorage.dm

```diff
@@ -0,0 +1,71 @@
+/datum/element/lazystorage
+
+/datum/element/lazystorage/Attach(datum/target)
+	. = ..()
+	if(!istype(target, /obj/item/storage))
+		return ELEMENT_INCOMPATIBLE
+	RegisterSignal(target, COMSIG_CONTAINS_STORAGE, .proc/on_check)
+	RegisterSignal(target, COMSIG_IS_STORAGE_LOCKED, .proc/check_locked)
+	RegisterSignal(target, COMSIG_TRY_STORAGE_SHOW, .proc/signal_show_attempt)
+	RegisterSignal(target, COMSIG_TRY_STORAGE_INSERT, .proc/signal_insertion_attempt)
+	RegisterSignal(target, COMSIG_TRY_STORAGE_CAN_INSERT, .proc/signal_can_insert)
+	RegisterSignal(target, COMSIG_TRY_STORAGE_TAKE_TYPE, .proc/signal_take_type)
+	RegisterSignal(target, COMSIG_TRY_STORAGE_FILL_TYPE, .proc/signal_fill_type)
+	RegisterSignal(target, COMSIG_TRY_STORAGE_SET_LOCKSTATE, .proc/set_locked)
+	RegisterSignal(target, COMSIG_TRY_STORAGE_TAKE, .proc/signal_take_obj)
+	RegisterSignal(target, COMSIG_TRY_STORAGE_QUICK_EMPTY, .proc/signal_quick_empty)
+	RegisterSignal(target, COMSIG_TRY_STORAGE_HIDE_FROM, .proc/signal_hide_attempt)
+	RegisterSignal(target, COMSIG_TRY_STORAGE_HIDE_ALL, .proc/close_all)
+	RegisterSignal(target, COMSIG_TRY_STORAGE_RETURN_INVENTORY, .proc/signal_return_inv)
+	RegisterSignal(target, COMSIG_CLICK_ALT, .proc/on_alt_click)
+	RegisterSignal(target, COMSIG_MOUSEDROP_ONTO, .proc/mousedrop_onto)
+	RegisterSignal(target, COMSIG_MOUSEDROPPED_ONTO, .proc/mousedrop_receive)
```

Comment:
Same here. You can register a list of signals to a single proctype.
That said, there is no need for a LAZYSTORAGE_PASSTHROUGH_PROC macro.

Reply:
The separate procs exist because signal handlers do not know which signal caused the proc to run, without it's not possible to call the right proc on the storage component

Reply:
Oh, thanks for telling us. I was too hasty on that review.